transmart-api-server/src/main/groovy/org/transmartproject/api/server/user/KeycloakUserResourceService.groovy

```diff
@@ -0,0 +1,136 @@
+package org.transmartproject.api.server.user
+
+import com.google.common.collect.ImmutableMultimap
+import com.google.common.collect.Multimap
+import groovy.util.logging.Slf4j
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.context.annotation.Primary
+import org.springframework.security.oauth2.provider.OAuth2Authentication
+import org.springframework.stereotype.Component
+import org.transmartproject.core.exceptions.NoSuchResourceException
+import org.transmartproject.core.users.AuthorisationChecks
+import org.transmartproject.core.users.ProtectedOperation
+import org.transmartproject.core.users.ProtectedResource
+import org.transmartproject.core.users.User
+import org.transmartproject.core.users.UsersResource
+
+import java.security.Principal
+import java.text.ParseException
+
+@Component
+@Primary
+@Slf4j
+class KeycloakUserResourceService implements UsersResource {
+
+    @Autowired
+    AuthorisationChecks authorisationChecks
+
+    @Override
+    User getUserFromUsername(String username) throws NoSuchResourceException {
+        throw new UnsupportedOperationException()
+    }
+
+    @Override
+    List<User> getUsers() {
+        throw new UnsupportedOperationException()
+    }
+
+    @Override
+    List<User> getUsersWithEmailSpecified() {
+        throw new UnsupportedOperationException()
+    }
+
+    @Override
+    User getUserFromPrincipal(Principal principal) {
+        assert principal instanceof OAuth2Authentication
+        def authentication = principal.userAuthentication
+        assert authentication: 'User is not authenticated.'
```

Comment:
Why not throw an exception (in `transmart-core-api`), to be caught by, e.g., [BusinessExceptionResolver](https://github.com/thehyve/transmart-core/blob/dev/transmart-rest-api/src/main/groovy/org/transmartproject/rest/http/BusinessExceptionResolver.groovy)?

Reply:
core-api should not know anything about spring security. But we could still throw `AccessDeniedException` from here and it should be caught by `BusinessExceptionResolver`